coffeecard/CoffeeCard.Library/Services/TokenService.cs

```diff
@@ -44,28 +44,73 @@ public JwtSecurityToken ReadToken(string token)
             return new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
         }
 
+        
         /// <summary>
-        ///     Receives the serialized version of a JWT token as a string. Checks if the JWT token is valid (Based on its lifetime
-        ///     and if it has been used before)
+        /// Receives the serialized version of a JWT token as a string.
+        /// Checks if the JWT token is valid (Based on its lifetime)
         /// </summary>
         /// <param name="tokenString"></param>
         /// <returns></returns>
-        public async Task<bool> ValidateToken(string tokenString)
+        public async Task<bool> ValidateTokenAsync(string tokenString)
         {
             try
             {
-                var tokenNotUsed = false;
+                var securityKey = new SymmetricSecurityKey(
+                    Encoding.UTF8.GetBytes(_identitySettings.TokenKey));
+
+                try
+                {
+                    var securityTokenHandler = new JwtSecurityTokenHandler();
+                    var validationParameters = new TokenValidationParameters
+                    {
+                        ValidateAudience = false,
+                        ValidateIssuer = false,
+                        ValidateIssuerSigningKey = true,
+                        IssuerSigningKey = securityKey,
+                        ValidateLifetime = true,
+                        ClockSkew = TimeSpan.Zero //the default for this setting is 5 minutes
+                    };
+
+                    securityTokenHandler.ValidateToken(tokenString, validationParameters, out _); // Throws exception if token is invalid
+                }
+                catch (Exception e) when (e is ArgumentException ||
+                                          e is SecurityTokenException)
+                {
+                    Log.Information("Received invalid token");
```

Comment:
For what reason would an ArgumentException be thrown and does it make sense to log the same message in this case? If it is unexpected to receive an ArgumentException, I think it's important to log the stack trace

Reply:
ArgumentException will be thrown if the token given for validation is malformed in some way, e.g., can't be read, is null, or if the validation parameters are null. I think it is safe to handle this in the same manner as an invalid token. 

I missed there being a second try catch surrounding this bit when I made it though, where it does catch ArgumentException, so we could also remove it from this catch and rely on that instead. Or remove the second one entirely, either way something needs to be changed.